### PR TITLE
[FSPE-6283] Add support of Vue, Stencil, Angular & React boilerplates

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Commit summary

- [x] Implemented Vue boilerplate
- [x] Implemented Stencil boilerplate
- [x] Automated zipping of Vue & Stencil boilerplates making them downloadable
- [x] Documented Framework boilerplates section on the Installation page
- [x] Implemented React boilerplate
- [x] Implement Angular boilerplate

https://ctl.atlassian.net/browse/FSPE-6283
https://ctl.atlassian.net/browse/FSPE-6324
https://ctl.atlassian.net/browse/FSPE-6325
https://ctl.atlassian.net/browse/FSPE-6336
https://ctl.atlassian.net/browse/FSPE-6338

### CODE REVIEW

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://11d99d3eee6b9c5c6e4e7405cff08cc607740340-chi-pr.enterprise.ctl.io" target="_blank"> https://11d99d3eee6b9c5c6e4e7405cff08cc607740340-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1340/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1341/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://0e7f111576700082ce456cc84bc5bdb37a951548-chi-pr.enterprise.ctl.io" target="_blank"> https://0e7f111576700082ce456cc84bc5bdb37a951548-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1342/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://00eeb80c6ae0d031d26f5724aa92d2789f788a54-chi-pr.enterprise.ctl.io" target="_blank"> https://00eeb80c6ae0d031d26f5724aa92d2789f788a54-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1343/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://f5f6ac1f1647709eb1941663a823a5912a6b1573-chi-pr.enterprise.ctl.io" target="_blank"> https://f5f6ac1f1647709eb1941663a823a5912a6b1573-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1352/ for more details.

**Review by SergioQCL — CHANGES_REQUESTED** [review]

File: src/boilerplates/react/react-boilerplate/src/components/header.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+class Header extends React.Component {
+	componentDidMount() {
+		chi.drawer(document.getElementById('drawer-portal-with-navbar'));
+		chi.dropdown(document.getElementById('button-site-portal-with-navbar'));
+		chi.dropdown(document.getElementById('button-user-portal-with-navbar'));
+		chi.dropdown(document.getElementById('button-eid-portal-with-navbar'));
+	}
+
+	render () {
```
[the remark this answers is not in the document]
Comment (janibolkvadze) [reply src/boilerplates/react/react-boilerplate/src/components/header.tsx]: Good catch, the formatting was wrong. I fixed the indentation and the other suggestions.
I also restructured the Vue boilerplate to make the structure consistent with the other Boilerplates.
Thank you!

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://72c42a4fca5cd045a47434eb9f186b6d7e6ff296-chi-pr.enterprise.ctl.io" target="_blank"> https://72c42a4fca5cd045a47434eb9f186b6d7e6ff296-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1430/ for more details.

**Comment by janibolkvadze** [comment]
test this please

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://fd70ccfb2620243d9dd9f53f37ce6c5cef143635-chi-pr.enterprise.ctl.io" target="_blank"> https://fd70ccfb2620243d9dd9f53f37ce6c5cef143635-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1433/ for more details.

**Review by SergioQCL — APPROVED** [review]
LGTM

File: src/website/views/getting-started/installation.pug [hunk 2]
```diff
@@ -178,6 +178,73 @@ p.-text.-mb--3
...
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        a(href='/boilerplates/vue-boilerplate.zip')
+          i.chi-icon.icon-circle-arrow-down-outline.-sm--2
```
Comment (mattnickles) [inline src/website/views/getting-started/installation.pug]:
```suggestion
        a.chi-button(href='/boilerplates/vue-boilerplate.zip')
          .chi-button__content
            i.chi-icon.icon-circle-arrow-down-outline
            span Download
```

File: src/website/views/getting-started/installation.pug [hunk 3]
```diff
@@ -178,6 +178,73 @@ p.-text.-mb--3
...
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        a(href='/boilerplates/stencil-boilerplate.zip')
+          i.chi-icon.icon-circle-arrow-down-outline.-sm--2
```
Comment (mattnickles) [inline src/website/views/getting-started/installation.pug]:
```suggestion
        a.chi-button(href='/boilerplates/stencil-boilerplate.zip')
          .chi-button__content
            i.chi-icon.icon-circle-arrow-down-outline
            span Download
```

File: src/website/views/getting-started/installation.pug [hunk 4]
```diff
@@ -178,6 +178,73 @@ p.-text.-mb--3
...
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        a(href='/boilerplates/angular-boilerplate.zip')
+          i.chi-icon.icon-circle-arrow-down-outline.-sm--2
```
Comment (mattnickles) [inline src/website/views/getting-started/installation.pug]:
```suggestion
        a.chi-button(href='/boilerplates/angular-boilerplate.zip')
          .chi-button__content
            i.chi-icon.icon-circle-arrow-down-outline
            span Download
```

File: src/website/views/getting-started/installation.pug [hunk 5]
```diff
@@ -178,6 +178,73 @@ p.-text.-mb--3
...
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        i.chi-icon.icon-check.-icon--success
+      td
+        a(href='/boilerplates/react-boilerplate.zip')
+          i.chi-icon.icon-circle-arrow-down-outline.-sm--2
```
Comment (mattnickles) [inline src/website/views/getting-started/installation.pug]:
```suggestion
        a.chi-button(href='/boilerplates/react-boilerplate.zip')
          .chi-button__content
            i.chi-icon.icon-circle-arrow-down-outline
            span Download
```

**Review by mattnickles — COMMENTED** [review]
Might be helpful to style these as a button and add an explicit text label

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://0f5862206ecf5c123e7c980066f5fc3dd9f99d64-chi-pr.enterprise.ctl.io" target="_blank"> https://0f5862206ecf5c123e7c980066f5fc3dd9f99d64-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1477/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://a3c5de3aeb6b2a64a38fd15f1cbca76777009eae-chi-pr.enterprise.ctl.io" target="_blank"> https://a3c5de3aeb6b2a64a38fd15f1cbca76777009eae-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1478/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Build was ABORTED. See http://jenkins.elasticbox.com/job/chi_pull_request/1493/ for more details.

**Comment by allyourbotsarebelongtous** [comment]
Your CHI Pull Request website is available for review at <a href="https://44642850fac408df9c8c73a68e940f8c5ba0f70e-chi-pr.enterprise.ctl.io" target="_blank"> https://44642850fac408df9c8c73a68e940f8c5ba0f70e-chi-pr.enterprise.ctl.io </a>

**Comment by allyourbotsarebelongtous** [comment]
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/1494/ for more details.

**Review by jllr — APPROVED** [review]

**Review by mattnickles — APPROVED** [review]

**Comment by allyourbotsarebelongtous** [comment]
The following instances are being terminated: https://ops.elasticbox.com/services/instances/i-go8kp7, https://ops.elasticbox.com/services/instances/i-8gsywa, https://ops.elasticbox.com/services/instances/i-7bo0qo, https://ops.elasticbox.com/services/instances/i-cwo8nv, https://ops.elasticbox.com/services/instances/i-yp8ptg, https://ops.elasticbox.com/services/instances/i-hhi3x7, https://ops.elasticbox.com/services/instances/i-fwg6gj, https://ops.elasticbox.com/services/instances/i-mx5ny2, https://ops.elasticbox.com/services/instances/i-09k1wv

**Pull request merged**